### ISSUE TITLE
Verify if kubeadmin can be a regular OAuth user with latest Keycloak and OpenShift

### ISSUE THREAD

**Issue by l0rd** [eclipse-che/che#16835]
### Is your enhancement related to a problem? Please describe.

We currently do not support using OpenShift OAuth when `kubeadmin` is the only user of the cluster. That's because we assume that Keycloak doesn't support it. But that's something we verfied 1 year ago. And things [may have changed](https://github.com/eclipse/che/issues/14013#issuecomment-623187601).

### Describe the solution you'd like

Verify if latest versions of Keycloak and OpenShift allow to use kubeadmin as a valid OAuth identity.

**Comment by davidfestal** [comment]
I justed tested on a 4.3.9 OpenShift cluster, and the JSON profile returned for the `kube:admin` user is the following:
```json
╰─ curl -k -H "Authorization: Bearer xxxxxxxxx" "https://api.test-ocp43.codereadyqe.com:6443/apis/user.openshift.io/v1/users/~"
{
  "kind": "User",
  "apiVersion": "user.openshift.io/v1",
  "metadata": {
    "name": "kube:admin",
    "selfLink": "/apis/user.openshift.io/v1/users/kube%3Aadmin",
    "creationTimestamp": null
  },
  "identities": null,
  "groups": [
    "system:authenticated",
    "system:cluster-admins"
  ]
}                                                                                                                                                                                                                                            
``` 

Note that there is no "identity", since the `kube:admin| user is a hard-coded internal user that is not linked to any OAuth identity provider: it exists by default before installing any identity provider.

As a comparison, here is the profile of a user created with the `htpasswd` identity provider:

```json
╰─ curl -k -H "Authorization: Bearer xxxxxxx" "https://api.test-ocp43.codereadyqe.com:6443/apis/user.openshift.io/v1/users/~"
{
  "kind": "User",
  "apiVersion": "user.openshift.io/v1",
  "metadata": {
    "name": "user",
    "selfLink": "/apis/user.openshift.io/v1/users/user",
    "uid": "e01e78aa-50d9-482e-8bb4-3442a5247313",
    "resourceVersion": "5462325",
    "creationTimestamp": "2020-04-17T09:47:59Z"
  },
  "identities": [
    "htpasswd:user"
  ],
  "groups": [
    "system:authenticated",
    "system:authenticated:oauth"
  ]
}
```

Now here is why the limitation is still there in the latest Keycloak: The `openshift-v4` Keycloak identity provider requires a non-null `uid` in order to be able to link the openshift user to the Keycloak user, as shown here:
- https://github.com/keycloak/keycloak/blob/master/services/src/main/java/org/keycloak/social/openshift/OpenshiftV4IdentityProvider.java#L94
- https://github.com/keycloak/keycloak/blob/master/server-spi-private/src/main/java/org/keycloak/broker/provider/BrokeredIdentityContext.java#L52

In the case of the user `kubeadmin`, the user profile metadata doesn't contain such a uid.
If we could either:
- return an uid in the user profile OS endpoint
- or use a hard-coded, kubeadmin-dedicated `uid` in the Keycloak code shown above,
I assume this could allow releasing this limitation from the Keycloak-Openshift integration.

To summarize: For now the `kubeadmin` user still cannot be linked to a Keycloak user with the `openshift-v4` Keycloak indentity provider. And it seems this can be fixed only by one of the 2 solutions mentioned above.

cc @slaskawi

**Comment by davidfestal** [comment]
On the Openshift side, I created the following issue: https://github.com/openshift/origin/issues/24950

**Comment by l0rd** [comment]
Closing this issue since the verification has been done.

**Comment by iam-veeramalla** [comment]
Just to update here ... This is now posible. 
@l0rd

**Comment by l0rd** [comment]
@iam-veeramalla this is possible since Che v7.42. If you are testing on CRW it will be possible starting with CRW 2.16.

**Comment by iam-veeramalla** [comment]
I am not using CRW. I was waiting for Keycloak to support this feature, I saw previously that you were also waiting for this feature like me. 

I just thought of updating that it's possible now but looks like you guys already knew it :)

**Comment by l0rd** [comment]
Oh sorry, I didn't know about Keycloak. What we did is that we removed it as a dependency.
